testing/tests/DevExpress.ui.widgets/treeViewParts/checkboxes.js

```diff
@@ -238,3 +239,21 @@ QUnit.test("Selection works correct with custom rootValue", function(assert) {
     assert.ok(nodes[0].items[0].selected, "item was selected");
     assert.strictEqual(nodes[0].selected, undefined, "item selection has undefined state");
 });
+
+
+['dataSource', 'items'].forEach((optionName) => {
+    QUnit.test(`Nodes selection works even with redirect keys in ${optionName} option`, function(assert) {
```

Comment:
1. "even with redirect keys" - most likely the 'redirect' word will not be used to find tests for loop/cycle in hierarchy and it's difficult to understand what does this test check from this caption.
  `Nodes selection works even with loop/cycle in ${optionName} option` can be better

2. 'Node selection works' - this test calls the 'selectItem' function only. It doesn't cover all 'node selection' scenarios for all configurations.
  `selectItem with loop/cycle in ${optionName} option` can be better

Reply:
Great idea.
Fixed https://github.com/DevExpress/DevExtreme/pull/10695/commits/faa3daeb0edecae305d15ed75e2a93d6464ed1d7